sonata/plugins/covers_lastfm.py

```diff
@@ -35,14 +35,17 @@ def on_cover_fetch(artist, album, on_save_cb, on_err_cb):
     opener.addheaders = [("User-Agent", make_user_agent())]
 
     # First, find the link to the master release of this album
-    search_url = "http://ws.audioscrobbler.com/2.0/?%s" % (
-        urllib.parse.urlencode({
-            "method": "album.getInfo",
-            "artist": artist,
-            "album": album,
-            "api_key": API_KEY,
-            "format": "json",
-        }))
+    album_not_given = album == "" or album == None
+    if album_not_given:
+        search_url = "http://ws.audioscrobbler.com/2.0/?%s" % ( urllib.parse.urlencode({ "method": "artist.getInfo", "artist":
```

Comment:
All of this could be greatly simplified:
 * define your URL endpoint once
 * define a base dictionary to hold the parameters
 * update the dictionary base on ``not album`` (no need for the extra variable, nor for the double check)
 * produce a final ``search_url`` based on the previous definitions

Please respect [PEP 8 formatting](https://www.python.org/dev/peps/pep-0008/) for this, also (related to spaces and size of the lines).